### PR TITLE
Adding annobin test cases

### PR DESCRIPTION
**Pull request by nasastry**
security/annobin-tests.py adding test cases of annobin.
A tool for checking the security hardening status of binaries

Signed-off-by: Nageswara R Sastry <rnsastry@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by nasastry** [comment]
```
JOB ID     : 51295854c069a1b23e7b9d69d68b5e76a9811f9b
JOB LOG    : /home/nasastry/avocado-fvt-wrapper/results/job-2020-09-29T11.46-5129585/job.log
 (1/1) annobin-tests.py:annobin.test: FAIL: 6 test(s) failed, please refer to the log (13.73 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/nasastry/avocado-fvt-wrapper/results/job-2020-09-29T11.46-5129585/results.html
JOB TIME   : 14.38 s
```

**Comment by nasastry** [comment]
Couldn't figure out archive/master.zip url for this repository. so cloning the repository using git in the test script.

File: security/annobin-tests.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    annobin-testsuite
+    :avocado: tags=security,testsuite
+    """
+    def setUp(self):
+        '''
+        Install the basic packages to support annobin
+        '''
+        # Check for basic utilities
+        smm = SoftwareManager()
+        deps = ['gcc', 'make', 'autoconf', 'gcc-plugin-devel', 'rpm-devel',
```
Comment (harish-24) [inline security/annobin-tests.py:34]: Is the package name `gcc-plugin-devel` same for SuSE as well? I did not find this. Also not sure if `rpm-devel` would be available in debian based OS. Better to handle/exclude the distros which the test is not run on

**Review by harish-24 — CHANGES_REQUESTED** [review]

**Comment by nasastry** [comment]
@harish-24 Taken care of Ubuntu

Ubuntu:
```
JOB ID     : 2b0f34134f92a1909bd95de56cf10f310ed7f704
JOB LOG    : /root/avocado/job-results/job-2020-10-05T10.09-2b0f341/job.log
 (1/1) annobin-tests.py:annobin.test: CANCEL: Ubuntu not supported for this test (0.00 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 1
JOB TIME   : 0.36 s
```

RHEL:
```
JOB ID     : d9acaf1e7b0b129bc4009770b5412f32b7cc35ad
JOB LOG    : /home/nasastry/avocado-fvt-wrapper/results/job-2020-10-05T11.10-d9acaf1/job.log
 (1/1) annobin-tests.py:annobin.test: FAIL: 5 test(s) failed, please refer to the log (16.30 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/nasastry/avocado-fvt-wrapper/results/job-2020-10-05T11.10-d9acaf1/results.html
JOB TIME   : 16.37 s
```

**Comment by harish-24** [comment]
@nasastry Can you paste a run log on SLES? I do not find `gcc-plugin-devel` on SuSE as mentioned in the previous comment as well.

**Comment by nasastry** [comment]
RHEL:
```
JOB ID     : a08d35ebfc03ea20ee647678e6586fcea0011423
JOB LOG    : /home/nasastry/avocado-fvt-wrapper/results/job-2020-10-06T02.56-a08d35e/job.log
 (1/1) annobin-tests.py:annobin.test: FAIL: 5 test(s) failed, please refer to the log (12.62 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/nasastry/avocado-fvt-wrapper/results/job-2020-10-06T02.56-a08d35e/results.html
JOB TIME   : 12.68 s
```

Ubuntu:
```
JOB ID     : fe15d55c0f95cec85895cd82b1bdbb65a5142aa9
JOB LOG    : /root/avocado/job-results/job-2020-10-06T01.56-fe15d55/job.log
 (1/1) annobin-tests.py:annobin.test: CANCEL: Ubuntu not supported for this test (0.00 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 1
JOB TIME   : 0.36 s
```

SLES:
```
JOB ID     : d4b72998ef076bfa9aeeb736db3da947894df9c5
JOB LOG    : /home/OpTest/avocado-fvt-wrapper/results/job-2020-10-09T23.08-d4b7299/job.log
 (1/1) annobin-tests.py:annobin.test: CANCEL: SuSE not supported for this test (0.73 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 1
JOB HTML   : /home/OpTest/avocado-fvt-wrapper/results/job-2020-10-09T23.08-d4b7299/results.html
JOB TIME   : 1.58 s
```

**Comment by nasastry** [comment]
@harish-24 SLES and Ubuntu has been taken care. Thanks!!

**Review by harish-24 — APPROVED** [review]

**Pull request merged**